x-pack/plugins/gis/public/shared/layers/sources/ems_tms_source/ems_tms_source.js

```diff
@@ -71,9 +71,18 @@ export class EMSTMSSource extends TMSSource {
   }
 
   _getTMSOptions() {
-    return this._emsTileServices.find(service => {
+    if(!this._emsTileServices || !this._emsTileServices.length) {
+      return;
+    }
+
+    const emsTmsService = this._emsTileServices.find(service => {
       return service.id === this._descriptor.id;
     });
+    if (!emsTmsService) {
+      console.error(`EMS TMS Service: ${this._descriptor.id} currently unavailable`);
```

Comment:
Not sure this console error is needed. Just going to spam the console. The server should log an error if it can not reach EMS. Not need for the console to log errors since there will be something visible to the user through the UI